fla/ops/path_attn/parallel.py

```
@@ -200,13 +199,16 @@ def parallel_path_attention(
     """
     if scale is None:
         scale = k.shape[-1]**-0.5
-    assert q.shape[-1] in [16, 32, 64], "only support head_dim in [16, 32, 64] for now. Stay tuned!"
-    assert v.shape[-1] in [16, 32, 64], "only support head_dim in [16, 32, 64] for now. Stay tuned!"
+    assert q.shape[-1] in [16, 32, 64, 128], "only support head_dim in [16, 32, 64, 128] for now. Stay tuned!"
+    assert v.shape[-1] in [16, 32, 64, 128], "only support head_dim in [16, 32, 64, 128] for now. Stay tuned!"
+    if w.dtype == torch.bfloat16:
+        w = w.to(torch.float16)
+        warnings.warn("Householder cumulative product with bfloat16 could be very inaccurate. We automatically convert w to float16 for better precision.")
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)